crates/model/src/solver_competition.rs

```diff
@@ -19,6 +21,8 @@ pub struct SolverCompetition {
     pub transaction_hash: Option<H256>,
     pub auction: CompetitionAuction,
     pub solutions: Vec<SolverSettlement>,
+    #[serde_as(as = "Option<BytesHex>")]
+    pub optimized_call_data: Option<Vec<u8>>,
```

Comment:
Why do we need to duplicate this calldata on the `SolverCompetition` object. Won't this always be at `competition.solutions[competition.solutions.length - 1].calldata`?

Reply:
I see - one other consideration is that the transaction calldata is stored on the blockchain, so it feels slightly redundant to include it here as well. Maybe I don't fully understand the motivation behind the PR (I think the description includes the "old way" the PR was implemented):

> I decided to clone the winning solution in case the optimization fails. This was done to have a consistent behavior where the second to last settlement is always the original winning solution and the last one is always the optimized winning solution (even if they are identical).

Reply:
Yes, the description is not up to date, yet. The motivation was that it is confusing when the data we store in the solver competition doesn't store what we actually submit on chain. I don't feel strongly for either option. I can simply revert the change only storing the call data again.